packages/block-editor/src/components/block-mover/test/index.native.js

```diff
@@ -46,4 +68,97 @@ describe( 'Block Mover Picker', () => {
 		const screen = render( <BlockMover { ...props } /> );
 		expect( screen.toJSON() ).toMatchSnapshot();
 	} );
+
+	describe( 'moving blocks', () => {
+		beforeAll( () => {
+			// Register all core blocks
+			registerCoreBlocks();
+		} );
+
+		afterAll( () => {
+			// Clean up registered blocks
+			getBlockTypes().forEach( ( block ) => {
+				unregisterBlockType( block.name );
+			} );
+		} );
+
+		it( 'moves blocks up and down', async () => {
+			const screen = await initializeEditor( {
+				initialHtml: INITIAL_HTML,
+			} );
+			const { getByLabelText } = screen;
+
+			// Get Spacer block
+			const spacerBlock = await getBlock( screen, 'Spacer', {
+				rowIndex: 2,
+			} );
+			fireEvent.press( spacerBlock );
+
+			// Tap on the Move Down button
+			const downButton = getByLabelText(
+				/Move block down from row 2 to row 3/
+			);
+			fireEvent.press( downButton );
+
+			// Get Heading block
+			const headingBlock = await getBlock( screen, 'Heading', {
+				rowIndex: 2,
+			} );
+			fireEvent.press( headingBlock );
+
+			// Tap on the Move Up button
+			const upButton = getByLabelText(
+				/Move block up from row 2 to row 1/
+			);
+			fireEvent.press( upButton );
+
+			expect( getEditorHtml() ).toMatchSnapshot();
+		} );
+
+		it( 'disables the Move Up button for the first block', async () => {
+			const screen = await initializeEditor( {
+				initialHtml: INITIAL_HTML,
+			} );
+			const { getByLabelText } = screen;
+
+			// Get Paragraph block
+			const paragraphBlock = await getBlock( screen, 'Paragraph' );
+			fireEvent.press( paragraphBlock );
+
+			// Get the Move Up button
+			const upButton = getByLabelText( /Move block up/ );
+			const isUpButtonDisabled =
+				upButton.props.accessibilityState?.disabled;
+			expect( isUpButtonDisabled ).toBe( true );
+
+			// Press the button to make sure the move doesn't block
+			fireEvent.press( upButton );
+
+			expect( getEditorHtml() ).toMatchSnapshot();
+		} );
+
+		it( 'disables the Move Down button for the last block', async () => {
+			const screen = await initializeEditor( {
+				initialHtml: INITIAL_HTML,
+			} );
+			const { getByLabelText } = screen;
+
+			// Get Heading block
+			const headingBlock = await getBlock( screen, 'Heading', {
+				rowIndex: 3,
+			} );
+			fireEvent.press( headingBlock );
+
+			// Get the Move Down button
+			const downButton = getByLabelText( /Move block down/ );
+			const isDownButtonDisabled =
+				downButton.props.accessibilityState?.disabled;
+			expect( isDownButtonDisabled ).toBe( true );
+
+			// Press the button to make sure the move doesn't block
```

Comment:
```suggestion
			// Press the button to make sure the block doesn't move
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/46699/commits/36fb048fd8c33fc816156171aee9f070432de20a

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/46699/commits/36fb048fd8c33fc816156171aee9f070432de20a